test/e2e/webdriver/driver.js

```diff
@@ -546,6 +546,43 @@ class Driver {
     throw new Error(`No window with title: ${title}`);
   }
 
+  async switchToWindowWithUrl(
+    url,
+    initialWindowHandles,
+    delayStep = 1000,
+    timeout = this.timeout,
+    { retries = 8, retryDelay = 2500 } = {},
+  ) {
+    let windowHandles =
+      initialWindowHandles || (await this.driver.getAllWindowHandles());
+    let timeElapsed = 0;
+
+    while (timeElapsed <= timeout) {
+      for (const handle of windowHandles) {
+        const handleUrl = await retry(
+          {
+            retries,
+            delay: retryDelay,
+          },
+          async () => {
+            await this.driver.switchTo().window(handle);
+            return await this.driver.getCurrentUrl();
+          },
+        );
+
+        if (handleUrl === `${url}/`) {
+          return handle;
+        }
+      }
```

Comment:
ahh i'm sorry to comment on this one again. It just seems weird to have what is essentially two retry/timeout loops. Is this inner `retry` necessary? 

timeElapsed isn't really the time elapsed since it doesn't account for how long the execution of the loop took

Reply:
I believe the retry is necessary, maybe the `while (timeElapsed <= timeout)` part might not be though. These methods have been combined to combat the flaky tests [PR](https://github.com/MetaMask/metamask-extension/pull/21394) and internal discussions as to the best approach [internal disc](https://consensys.slack.com/archives/GTQAGKY5V/p1697223609357109?thread_ts=1697150920.731179&cid=GTQAGKY5V)  and an open issue https://github.com/MetaMask/metamask-extension/issues/21403 for future refactoring in the future.

Reply:
Thinking more largely, we could combine all the switching windows methods together `switchToWindow`, `switchToWindowWithTitle`, and `switchToWindowWithUrl`.

Reply:
Thank you for explaining 🙏